starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/GradleBuildCache.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.build.gradle;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.gradle.GradleDsl;
+import io.micronaut.starter.build.gradle.GradlePlugin;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.build.gradle.templates.buildCache;
+import io.micronaut.starter.template.RockerWritable;
+
+public abstract class GradleBuildCache implements Feature, BuildRemoteCacheConfiguration {
```

Comment:
I am confused. You're not overriding `isVisible()`, which defaults to `true`. So how does this feature gets hidden?

Reply:
I am overriding it in `MicronautBuildCache` which is the only `@Singleton` implementation. `GradleBuildCache` is an abstract class.